js/game.js

```diff
@@ -0,0 +1,253 @@
+        this.makeMove(1,1)
+      }else if (random > 0.75){
+        this.makeMove(2,2)
+      }else if (random > 0.5) {
+        this.makeMove(2, 0)
+      }else if (random > 0.25) {
+        this.makeMove(0, 2)
+      }else {
+        this.makeMove(0, 0)
+      }
+    }
+    else{
+      let cases = [];
+
+      const computer = this.activePlayer;
+      const human = this.unactivePlayer;
+
+      let allFields = [];
+      this.board.fields.forEach(field => allFields = [...allFields, ...field]);
+
+      const avaliableFields = allFields.filter(field => field.free);
+      avaliableFields.forEach(field => {
+        field.owner = computer;
+        field.free = false;
+        let step1 = Object.assign({}, { x: field.x, y: field.y })
+        if (this.checkForWin(computer)) {
+          step1.score = 10;
+          cases = [...cases, Object.assign({}, step1)];
+        }else {
+          step1.score = 0;
+          cases = [...cases, Object.assign({}, step1)];
+          const avaliableFields = allFields.filter(field => field.free);
+          avaliableFields.forEach(field => {
+            field.owner = human;
+            field.free = false;
+            if (this.checkForWin(human)) {
+              avaliableFields.forEach(() => {
+                step1.score = -10;
+                cases = [...cases, Object.assign({}, step1)];
+              });
+            } else {
+              step1.score = 0;
+              cases = [...cases, Object.assign({}, step1)];
+              const avaliableFields = allFields.filter(field => field.free);
+              avaliableFields.forEach(field => {
+                field.owner = computer;
+                field.free = false;
+                if (this.checkForWin(computer)) {
+                  step1.score = 9;
+                  cases = [...cases, Object.assign({}, step1)];
+                } else {
+                  step1.score = 0;
+                  cases = [...cases, Object.assign({}, step1)];
+                  const avaliableFields = allFields.filter(field => field.free);
+                  avaliableFields.forEach(field => {
+                    field.owner = human;
+                    field.free = false;
+                    if (this.checkForWin(human)) {
+                      avaliableFields.forEach(() => {
+                        step1.score = -9;
+                        cases = [...cases, Object.assign({}, step1)];
+                      });
+                    } else {
+                      step1.score = 0;
+                      cases = [...cases, Object.assign({}, step1)];
+
+                    }
+                    field.owner = {};
+                    field.free = true;
+                  });
+                }
+                field.owner = {};
+                field.free = true;
+              });
+            }
+            field.owner = {};
+            field.free = true;
+          });
+        }
+        field.owner = {};
+        field.free = true;

```

Comment:
Cała reszta kodu była ładna i dość czytelna ale tutaj poleciałeś ;) Wpadasz tutaj w `if / else hell` i generalnie niemal `callback hell`. To nie może wyjść w tej formie. Po pierwsze podziel to na mniejsze funkcje, po drugie postaraj się ograniczyć ilość `if / else`. Generalnie to jest całe do przerobienia

Reply:
Więcej jutro jak drugi raz to przeczytam i zastanowie się jak to uprościć

Reply:
Tutaj, nie mam absoultnie pomyslu. teraz Wiem jak algorytm powinien myslec, ale przerasta mnei napisanie tego. na chwile obecna sprawdzamy kazda kombinacje na przod. jezeli by liczyc od pierwszego kroku to wszystkich kombinacji jest 40 320... dlatego zaczynam od drugiego ruchu komputera, bo wtedy kombinacji do przleiczenia jest 720, a pierwszy ruch nie ma takiego znaczenia. nie moze tylko zostac wykonany w polach [1,0] [0,1] [2,1] i [1,2]. ale nad tym siedizalem poltora dnia i nie jestem w stanie wymyslic nic wiecej (a i tak to co tu jest nie działa dobrze ;) )